src/components/not-found/NotFound.tsx

```diff
@@ -1,32 +1,26 @@
-import autobind from "autobind-decorator";
 import * as React from "react";
-import { RouteComponentProps, withRouter } from "react-router-dom";
+import { useHistory } from "react-router-dom";
 import { i18n } from "../../Locale";
 import { MAIN_MENU } from "../../routes";
 import Button from "../shared/button/Button";
 import "./NotFound.scss";
 
-type Props = RouteComponentProps<any>;
+function NotFound() {
+  const history = useHistory();
 
-@autobind
-class NotFound extends React.Component<Props> {
-  public render() {
-    return (
-      <div className="not-found">
-        <span className="not-found__404">404</span>
+  const goHome = React.useCallback(() => history.push(MAIN_MENU), [history]);
 
-        <h1 className="not-found__text">{i18n("not_found_text")}</h1>
+  return (
+    <div className="not-found">
+      <span className="not-found__404">404</span>
 
-        <Button className="not-found__cta" onClick={this.goHome}>
-          {i18n("home")}
-        </Button>
-      </div>
-    );
-  }
+      <h1 className="not-found__text">{i18n("not_found_text")}</h1>
 
-  private goHome() {
-    this.props.history.push(MAIN_MENU);
-  }
+      <Button className="not-found__cta" onClick={goHome}>
```

Comment:
Another ignorable comment about a button that should be a link.

Reply:
Yes. Will be taken care once base component will be refactored.